lib/game_teams_manager.rb

```diff
@@ -73,30 +73,60 @@ def winningest_coach(season_games)   ##Refactor?: change 'hash' to 'coach'
   end
 
   def worst_coach(season_games)   ##Refactor?:  change 'hash' to 'coach'
-    hash = Hash.new { |hash, team| hash[team] = [0,0] }
+    coach = Hash.new { |hash, team| hash[team] = [0,0] }
     @game_teams.each do |game_team|
       if season_games.include?(game_team.game_id)
-        hash[game_team.head_coach][1] += 1
-        hash[game_team.head_coach][0] += 1 if game_team.result == "WIN"
+        coach[game_team.head_coach][1] += 1
+        coach[game_team.head_coach][0] += 1 if game_team.result == "WIN"
       end
     end
-    hash.each do |team_id, pair|
+    coach.each do |team_id, pair|
       ratio = calculate_ratios(pair)
-      hash[team_id] = ratio
+      coach[team_id] = ratio
     end
-    hash.key(hash.values.min)
+    coach.key(coach.values.min)
   end
 
-#get_all_game_numbers_a_team_played_in
-#if game number == team_game.number && team_id != team_game.team_id
-# => if win, += 1 to hash wins, else += to hash total_games
-#analize the hash numbers, to get a hash with ratios
-#analize the values to find the bigest/smallest
-#return key that associated with biggest or smallest value
+  def favorite_opponent(team_id)
```

Comment:
This and rival look like we can break into separate helpers that will work for both